File: src/main/java/com/currencies/mainpackage/core/ticket/TicketService.java
package com.currencies.mainpackage.core.ticket;

import com.currencies.mainpackage.api.dto.response.TicketResponse;
import com.currencies.mainpackage.api.dto.request.CreateTicketRequest;
import org.jetbrains.annotations.NotNull;

import java.sql.Date;
import java.util.List;

public interface TicketService {

    @NotNull
    List<TicketResponse> findAll();

    @NotNull
    TicketResponse findById(@NotNull Integer id);

    @NotNull
    List<TicketResponse> findTickets(String from, String to, Date startFlightDate);

    @NotNull
    TicketResponse createTicket(@NotNull CreateTicketRequest request);

    @NotNull
    TicketResponse update(@NotNull Integer id, @NotNull CreateTicketRequest request);

    void delete(@NotNull Integer id);


}

File: src/main/java/com/currencies/mainpackage/core/ticket/TicketService.kt
package com.currencies.mainpackage.core.ticket

import com.currencies.mainpackage.api.dto.request.CreateTicketRequest
import com.currencies.mainpackage.api.dto.response.TicketResponse
import java.sql.Date

interface TicketService {

    fun findAll(): List<TicketResponse>

    fun findById(id: Long): TicketResponse?

    fun findTickets(from: String, to: String, startFlightDate: Date): List<TicketResponse>

    fun save(request: CreateTicketRequest): TicketResponse

    fun update(id: Long, request: CreateTicketRequest): TicketResponse

    fun delete(id: Long)

}
